casanovo/denovo/model.py

```
@@ -1293,55 +1300,67 @@ def _psm_batches(
+            total=len(candidates),
+            desc="Scoring candidates",
+            unit="PSM",
+            leave=False,
+        )
+        while candidates:
+            batch_candidates = candidates[:batch_size]
+
+            # Repeat the spectrum information for each candidate to be matched.
+            psm_batch = {key: [] for key in [*batch.keys(), "seq"]}
+            for spec_i, cand in batch_candidates:
+                for key in batch.keys():
```

Ruff (0.15.6)

[warning] Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)